public/app/features/dashboard-scene/edit-pane/useEditableElement.ts

```diff
@@ -0,0 +1,31 @@
+import { useMemo } from 'react';
+
+import { SceneObject, VizPanel } from '@grafana/scenes';
+
+import { DashboardScene } from '../scene/DashboardScene';
+import { EditableDashboardElement, isEditableDashboardElement } from '../scene/types';
+
+import { DashboardEditableElement } from './DashboardEditableElement';
+import { VizPanelEditableElement } from './VizPanelEditableElement';
+
+export function useEditableElement(sceneObj: SceneObject | undefined): EditableDashboardElement | undefined {
```

Comment:
An idea - what if this hook returned a ref that one can use in the component for the main DOM element, and the hook would apply the inline selection styles, so that each individual component does not have to care about applying relevant class name? Think that would make it easier to re-use.

Reply:
@dprokop I don't understand how that would work, this hook has no connection to the dom element  and is not used anywhere where it could access / control the ref of the dom element being selected